arbnode/resourcemanager/resource_management_test.go

```diff
@@ -6,52 +6,57 @@ package resourcemanager
 import (
 	"fmt"
 	"os"
+	"regexp"
 	"testing"
 )
 
-func updateFakeCgroupv1Files(c *cgroupsV1MemoryLimitChecker, limit, usage, inactive int) error {
-	limitFile, err := os.Create(c.limitFile)
+func updateFakeCgroupFiles(c *cgroupsMemoryLimitChecker, limit, usage, inactive int) error {
+	limitFile, err := os.Create(c.files.limitFile)
 	if err != nil {
 		return err
 	}
-	_, err = fmt.Fprintf(limitFile, "%d\n", limit)
-	if err != nil {
+	if _, err = fmt.Fprintf(limitFile, "%d\n", limit); err != nil {
 		return err
 	}
 
-	usageFile, err := os.Create(c.usageFile)
+	usageFile, err := os.Create(c.files.usageFile)
 	if err != nil {
 		return err
 	}
-	_, err = fmt.Fprintf(usageFile, "%d\n", usage)
-	if err != nil {
+	if _, err = fmt.Fprintf(usageFile, "%d\n", usage); err != nil {
 		return err
 	}
 
-	statsFile, err := os.Create(c.statsFile)
+	statsFile, err := os.Create(c.files.statsFile)
 	if err != nil {
 		return err
 	}
-	_, err = fmt.Fprintf(statsFile, `total_cache 1029980160
+	if _, err = fmt.Fprintf(statsFile, `total_cache 1029980160
 total_rss 1016209408
 total_inactive_file %d
 total_active_file 321544192
-`, inactive)
-	if err != nil {
+`, inactive); err != nil {
 		return err
 	}
 	return nil
 }
 
-func TestCgroupsv1MemoryLimit(t *testing.T) {
+func TestCgroupsMemoryLimit(t *testing.T) {
 	cgroupDir := t.TempDir()
-	c := newCgroupsV1MemoryLimitChecker(cgroupDir, 95)
+	testFiles := cgroupsMemoryFiles{
+		limitFile:  cgroupDir + "/memory.limit_in_bytes",
+		usageFile:  cgroupDir + "/memory.usage_in_bytes",
+		statsFile:  cgroupDir + "/memory.stat",
+		inactiveRe: regexp.MustCompile(`total_inactive_file (\d+)`),
+	}
+
+	c := newCgroupsMemoryLimitChecker(testFiles, 95)
 	_, err := c.isLimitExceeded()
 	if err == nil {
 		t.Error("Should fail open if can't read files")
 	}
 
-	err = updateFakeCgroupv1Files(c, 1000, 1000, 51)
+	err = updateFakeCgroupFiles(c, 1000, 1000, 51)
 	if err != nil {
 		t.Error(err)
```

Comment:
When errors are not nil, there's probably no use of continuing the test, so this should be Fatal instead.

Same below/up.